Create schedule application

joncasdam opened the pull request:
Schedule application created inside conference app.

Fix for #1

joncasdam committed:
Create schedule application

Schedule application created inside conference app.

yakky commented on conference/schedule/admin.py:
```diff
@@ -0,0 +1,12 @@
+
+
+@admin.register(Talk)
+class SubmissionAdmin(ExportActionModelAdmin):
```
Maybe renaming this to "TalkAdmin"?

joncasdam replied:
fix on the way

yakky commented on conference/schedule/models.py:
```diff
@@ -0,0 +1,32 @@
+class TalkManager(models.Manager):
+    def get_queryset(self):
+        return super(TalkManager, self).get_queryset()\
+            .select_related(
```
Doesn't `select_related('talk')` suit? You just need to declare the linked model

joncasdam replied:
indeed.

yakky reviewed (commented):
I added a couple of comments

yakky commented:
@joncasdam thanks for this.
If you're willing to make a step forward with this applications we can add:
- Model additions
  - day: date field
  - start: timefield
  - end: timefield
  
  fields to print the start and end time of the talk in the schedule
- Figure out a solution to include slots in the schedule for non-talk items (like coffee breaks, lunch etc)

joncasdam committed:
Create action to approve submission via admin

Custom action to set as 'selected' submissions via admin.

ulope commented on conference/schedule/models.py:
```diff
@@ -14,6 +14,9 @@ def get_queryset(self):
     talk = models.ForeignKey(Submission, related_name='talks', limit_choices_to={'selected': True})
+    day = models.DateField(_('Date'), null=True, blank=True)
+    start = models.TimeField(_('Start'), null=True, blank=True)
+    end = models.TimeField(_('End'), null=True, blank=True)
```
It might make sense to replace the `end` field with a `duration` field. The advantages are that durations are trivially aggregratable (e.g. how many hour's of talks are there per day or some such) and also that in context of a conference talk duration seems more natural.

yakky replied:
It seems perfectly reasonable

joncasdam replied:
agreed. I'll implement it

joncasdam commented:
@yakky I've updated, squashed migrations and cleaned git story.
- renamed from `Talk` to `Slot` since it now can handle more than just talks
- replaced `end` field for `duration`
- integrated with latest changes in master
- added some properties and validations

yakky commented:
@joncasdam \o/ thanks for your effort!
